aviatrix/resource_aviatrix_gateway.go

```diff
@@ -412,6 +412,18 @@ func resourceAviatrixGateway() *schema.Resource {
 				ForceNew:    true,
 				Description: "Name of storage account with gateway images. Only valid for Azure China (2048)",
 			},
+			"enable_spot_instance": {
+				Type:         schema.TypeBool,
+				Optional:     true,
+				Description:  "Enable spot instance. NOT supported for production deployment.",
+				RequiredWith: []string{"spot_price"},
+			},
+			"spot_price": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Price for spot instance. NOT supported for production deployment.",
+				RequiredWith: []string{"enable_spot_instance"},
+			},
```

Comment:
If I set `enable_spot_instance` to false without setting `spot_price`, will this trigger an error?

Reply:
Yes. It is as expected. If customer is not using it, they should not set it (whether true or false), because it is optional and does not have a default value.